contracts/swap/swap.go

```diff
@@ -152,6 +180,34 @@ func (s simpleContract) PaidOut(opts *bind.CallOpts, addr common.Address) (*big.
 	return s.instance.PaidOut(opts, addr)
 }
 
+// TotalDeposit iterates over all Deposit events and returns the total amount ever deposited
+func (s simpleContract) TotalDeposit() (totalDeposit *big.Int, err error) {
+	depositIterator, err := s.instance.FilterDeposit(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for depositIterator.Next() {
+		totalDeposit = totalDeposit.Add(totalDeposit, depositIterator.Event.Amount)
+	}
+
+	return totalDeposit, depositIterator.Error()
+}
+
+// TotalWithdrawn iterates over all Withdraw events and returns the total amount ever withdrawn
+func (s simpleContract) TotalWithdrawn() (totalWithdrawn *big.Int, err error) {
+	withdrawIterator, err := s.instance.FilterWithdraw(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	for withdrawIterator.Next() {
+		totalWithdrawn = totalWithdrawn.Add(totalWithdrawn, withdrawIterator.Event.Amount)
+	}
+
+	return totalWithdrawn, withdrawIterator.Error()
```

Comment:
same thing as in `TotalDeposit` applies here

Reply:
Ok!

Reply:
`return totalWithdrawn, nil`